### PR TITLE
Fix grammar errors, improve documentation

### PR DESCRIPTION
**Pull request by jdpedrie**
Sorry for the flood of pull requests. I resolved a couple of spelling errors in the more recent changes, fixed a couple errors I introduced in my earlier changes to the documentation, and improved the overall wording of the documentation files.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,23 +52,17 @@ produce but is invaluable when present.
...
 
-  1. Code changes - typically in files located in `src/`
-  2. Unit tests to support - typically in files located in `test/`
+  1. Code changes - typically located in `src/`
+  2. Supporting unit tests - typically located in `test/`
   3. Version bump: major, minor, patch - in `package.json`
-  4. Following [commit message guidelines](http://www.tpope.net/node/106)
 
-Second commit should include:
+Second commit should include the output of:
```
Comment (kalisjoshua) [inline CONTRIBUTING.md]: What about "A second commit, if necessary, should include the output of:"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -52,23 +52,17 @@ produce but is invaluable when present.
...
 
   1. `npm run doc`
   2. `npm run coverage`
```
Comment (kalisjoshua) [inline CONTRIBUTING.md]: I feel that something about why the two commits should be kept separate - namely, diffs and code review - is still good to have.

**Comment by kalisjoshua** [comment]
I appreciate the attention to detail and helping out with this. Other than my two comments this is looking better and better. What you think about my comments?

**Comment by jdpedrie** [comment]
Can the `npm run doc` and `npm run coverage` be integrated into a travis build?

I had a really great reply all typed up, but I accidentally clicked off the page and it disappeared. :(

The gist of it was that I'd prefer we err on the side of giving discretion to contributors and reviewers. Rather than proscribe a precise two-commit pull request flow, offer general guidelines for what is expected and what is suggested. So we require test coverage and passing tests, and we suggest updates to the sample RAML and documentation. Additional guidelines or restrictions may arise naturally as we see how contributions work in practice.

**Comment by jdpedrie** [comment]
Something like this is what I'm referring to with travis: https://github.com/steveklabnik/automatically_update_github_pages_with_travis_example

**Comment by kalisjoshua** [comment]
Travis, or something like it, is going to happen. I just haven't had time to dedicate to it.

What do you think about my comments to this PR?

**Comment by jdpedrie** [comment]
I agree with them as far as they go, however, like I said in my 9:24 comment, I would prefer a different approach to what we expect from contributions.

This might be something to discuss internally, so we can establish guidelines for all the stuff QL open sources? Might make more sense than a patchwork of `CONTRIBUTING.md` files across our organization.

**Comment by jdpedrie** [comment]
pinging @kalisjoshua -- updated per our hipchat discussions.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -52,23 +52,18 @@ produce but is invaluable when present.
...
-
-  1. `npm run doc`
-  2. `npm run coverage`
-
-Documenation is generated by a tool, so checking a diff of generated files is,
-to some degree, pointless
-
-Keeping these seperations in a Pull Request will help keep code reviews short.
+Please avoid working directly on `master`. Feature branches are considered
+preferable, to avoid falling behind HEAD in your fork, and to ensure logical
```
Comment (kalisjoshua) [inline CONTRIBUTING.md]: HEAD should be master in this paragraph; HEAD is just the tip of the current branch.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -52,23 +52,18 @@ produce but is invaluable when present.
 
 ## Clean Pull Requests
 
-In an effort to keep the history clean, Pull Requests should be as concise as
-possible. This means consolodating changes into logical groups; change to the
-project, and documenation updates.
+In an effort to keep maintain a clean history, Pull Requests should be as concise as
```
Comment (kalisjoshua) [inline CONTRIBUTING.md]: I think the beginning of this sentence is not what you intended.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -17,13 +18,12 @@ The basics are:
 When adding a new rule, the definition of how it should perform needs to be
 documented and one of the best ways of doing that is a test. This test will
 become a part of the repository and will be run any time the code changes. The
-test passing will provide assurance to any future editors that any changes do
+passing test will provide assurance to any future editors that any changes do
```
Comment (kalisjoshua) [inline CONTRIBUTING.md]: test(s)

**Pull request merged**